language/coalescer.go

```diff
@@ -0,0 +1,70 @@
+// Package language implements structs and functions common between
+// programming language indexing implementation.
+package language
+
+import (
+	"context"
+
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/indexer"
+)
+
+var _ indexer.Coalescer = (*coalescer)(nil)
+
+type coalescer struct{}
+
+// NewCoalescer returns a new common programming language coalescer.
+func NewCoalescer(_ context.Context) (indexer.Coalescer, error) {
+	return &coalescer{}, nil
+}
+
+// Coalesce implements [indexer.Coalescer].
```

Comment:
The logic here is actually not trivial w/r/t which packages end up in the final index report, I think this docstring could do with more explanation.